teleport/tests/test_integration.py

```diff
@@ -0,0 +1,25 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import pytest
+
+from datadog_checks.teleport import TeleportCheck
+
+pytestmark = [pytest.mark.integration, pytest.mark.usefixtures('dd_environment')]
+
+
+def test_connect_ok(aggregator, dd_run_check):
+    instance = {"diagnostic_url": "http://127.0.0.1:3000"}
+    check = TeleportCheck('teleport', {}, [instance])
+    dd_run_check(check)
+    aggregator.assert_service_check('teleport.health.up', status=TeleportCheck.OK, count=1)
+    aggregator.assert_service_check('teleport.health.up', status=TeleportCheck.CRITICAL, count=0)
+
+
+def test_check_collects_teleport_common_metrics(aggregator, dd_run_check):
+    instance = {"diagnostic_url": "http://127.0.0.1:3000"}
+    check = TeleportCheck('teleport', {}, [instance])
+    dd_run_check(check)
+
+    aggregator.assert_metric("teleport.process_state")
```

Comment:
Is there only one common metric?

Reply:
No, but this is the metric that we're pretty sure will be generated as soon as the container starts. And since this is an integration test, we don't really need to assert all the metrics.

Reply:
So I guess we will create unit tests to verify that all metrics are reported correctly when the configuration indicates so.

Reply:
Example: https://github.com/DataDog/integrations-core/pull/16877/files#diff-73d0556e9d3e99aa91528fa5cf7aade9d213ec9bfe7ed877a4b2c36f136a44ceR22-R43